packages/lib/src/components/Card/components/CardInput/components/DisclaimerMessage/DisclaimerMessage.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { shallow } from 'enzyme';
+import { h } from 'preact';
+import DisclaimerMessage from './DisclaimerMessage';
+
+describe('DisclaimerMessage', () => {
+    const disclaimerMessage = {
+        message: 'By continuing you accept the %{linkText} of MyStore',
+        linkText: 'terms and conditions',
+        link: 'https://www.adyen.com'
+    };
+
+    const getWrapper = prop => shallow(<DisclaimerMessage disclaimer={prop} />);
+
+    test('Renders the DisclaimerMessage with text before and after the link', () => {
+        const wrapper = getWrapper(disclaimerMessage);
+        expect(wrapper.find('.adyen-checkout-disclaimer__label')).toHaveLength(1);
+        expect(wrapper.find('.adyen-checkout-disclaimer__label').text()).toContain('By continuing you accept the terms and conditions of MyStore');
+
+        expect(wrapper.find('.adyen-checkout__link')).toHaveLength(1);
+        expect(wrapper.find('.adyen-checkout__link').text()).toContain('terms and conditions');
+
+        expect(wrapper.find('[href="https://www.adyen.com"]')).toHaveLength(1);
```

Comment:
Just FYI, by using react testing library here, you could focus more on testing the Component as a whole rather than the implementation details (like checking classnames). It also makes the test easier to read IMO.

```js
test('Renders the DisclaimerMessage with text before and after the link', () => {
    render(<DisclaimerMessage disclaimer={disclaimerMessage} />);    
    expect(screen.getByText('By continuing', { exact: false }).textContent).toEqual(
            'By continuing you accept the terms and conditions of MyStore'
        );
    expect(screen.getByRole('link', { name: 'terms and conditions' })).toHaveAttribute('href', 'https://www.adyen.com');
});
```

Reply:
will do